builder/config.go

```diff
@@ -158,3 +155,47 @@ func (config *MixConfig) LoadConfig(filename string) error {
 
 	return nil
 }
+
+func (config *MixConfig) expandEnv() error {
+	re := regexp.MustCompile(`\$\{?([[:word:]]+)\}?`)
+	rc := reflect.ValueOf(config).Elem()
+
+	for i := 0; i < rc.NumField(); i++ {
+		s := rc.Field(i).Addr().Interface()
+		section := reflect.Indirect(reflect.ValueOf(s))
```

Comment:
Nop, it doesn't work. Without all this code you get an object that is readable, but not writable.

Reply:
Ah, right, because you created `rc` with `reflect.ValueOf(config).Elem()`, so it's not "settable". If you instead were to create it with the original reference (`reflect.ValueOf(config)`), you'd have a access to settable `Value`'s.

Reply:
Nop, also doesn't work. The problem here is not to get settable references to the members of the base structure, but to get settable references to the substructures within.

Reply:
Actually, it turns out that reflect.ValueOf(config).Elem() allows for settable substructures while reflect.ValueOf(*config) does not. After some testing with different combinations I managed to scrap some of the convoluted reflection code.